File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if not head:
            return None
        
        iterNode = head
        head = head.next
        iterNode.next = None

        while head:
            temp = head
            head = head.next
            temp.next = iterNode
            iterNode = temp

        return iterNode

File: 0206-reverse-linked-list/0206-reverse-linked-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reverseList(head *ListNode) *ListNode {
    if head == nil {
        return nil
    }

    cur := head
    var prev *ListNode
    for cur != nil {
        cur.Next, cur, prev = prev, cur.Next, cur
    }

    return prev
}